File: Parking Lot/parking_lot_demo.py
from parking_lot import ParkingLot
from level import Level
from car import Car
from motorcycle import Motorcycle
from truck import Truck

class ParkingLotDemo:
    def run():
        parking_lot = ParkingLot()
        parking_lot.add_level(Level(1,1))
        parking_lot.add_level(Level(2,2))

        car1 = Car("10101")
        car2 = Car("11011")
        motorcycle = Motorcycle("20202")
        truck = Truck("30303")

        parking_lot.park_vehicle(car1)
        parking_lot.park_vehicle(motorcycle)
        parking_lot.park_vehicle(truck)
        parking_lot.park_vehicle(car2)

        parking_lot.display_availability()

        parking_lot.unpark_vehicle(car1)

        parking_lot.display_availability()


if __name__ == "__main__":
    ParkingLotDemo.run()

File: ParkingLot/ParkingLotDemo.cpp
#include "ParkingLot.h"
#include "Level.h"
#include "Car.h"
#include "Motorcycle.h"
#include "Truck.h"

int main() {
    ParkingLot* parkingLot = ParkingLot::getInstance();
    parkingLot->addLevel(Level(1, 1, 1, 1)); // 50 car spots, 30 motorcycle spots, 20 truck spots
    parkingLot->addLevel(Level(2, 5, 3, 2)); // 50 car spots, 30 motorcycle spots, 20 truck spots

    Car car("ABC123");
    Truck truck("XYZ789");
    Motorcycle motorcycle("M1234");
    Car car2("ABC124");

    // Park vehicles
    parkingLot->parkVehicle(&car);
    parkingLot->parkVehicle(&truck);
    parkingLot->parkVehicle(&motorcycle);
     parkingLot->parkVehicle(&car2);

    // Display availability
    parkingLot->displayAvailability();

    // Unpark vehicle
    parkingLot->unparkVehicle(&motorcycle);
    parkingLot->unparkVehicle(&car2);

    // Display updated availability
    parkingLot->displayAvailability();

    return 0;
}